src/main/kotlin/net/minecrell/pluginyml/paper/PaperPluginDescription.kt

```diff
@@ -56,28 +57,34 @@ class PaperPluginDescription(project: Project) : PluginDescription() {
     @Input @Optional @JsonProperty("has-open-classloader") var hasOpenClassloader: Boolean? = null
     @Input @Optional @JsonProperty("folia-supported") var foliaSupported: Boolean? = null
 
-    @Nested @Optional @JsonProperty("dependencies") @JsonSerialize(`as` = Collection::class)
-    var depends: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
-    @Nested @Optional @JsonProperty("load-before") @JsonSerialize(`as` = Collection::class)
-    var loadBefore: NamedDomainObjectContainer<LoadDefinition> = project.container(LoadDefinition::class.java)
-    @Nested @Optional @JsonProperty("load-after") @JsonSerialize(`as` = Collection::class)
-    var loadAfter: NamedDomainObjectContainer<LoadDefinition> = project.container(LoadDefinition::class.java)
+    @Nested @Optional @JsonIgnore
+    var serverDependencies: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
+    @Nested @Optional @JsonIgnore
+    var bootstrapDependencies: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
+
+    @JsonProperty("dependencies") @JsonSerialize(contentAs = Collection::class)
+    fun dependencies(): Map<String, NamedDomainObjectContainer<DependencyDefinition>> = mapOf(
```

Comment:
```suggestion
    internal fun dependencies(): Map<String, NamedDomainObjectContainer<DependencyDefinition>> = mapOf(
```

(if that works, but that's what I had for something like this at some point https://github.com/Minecrell/plugin-yml/blob/d90424f2216864cf71a492b48fab9dc500cd95b4/src/main/kotlin/net/minecrell/pluginyml/bukkit/BukkitPluginDescription.kt#L56-L59)

Reply:
Changing it to `internal` made Kotlin leak its abstractions all over the place. It changes the name of it from `dependencies` to `dependencies$plugin_yml`.